spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+              "When enabled, the josdk metrics will be added in metrics source and "
+                  + "configured for operator.")
+          .defaultValue(true)
+          .build();
+
+  public static final ConfigOption<Boolean> KUBERNETES_CLIENT_METRICS_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.clientMetricsEnabled")
+          .defaultValue(true)
+          .description(
+              "Enable KubernetesClient metrics for measuring the HTTP traffic to "
+                  + "the Kubernetes API Server. Since the metrics is collected "
+                  + "via Okhttp interceptors, can be disabled when opt in "
+                  + "customized interceptors.")
+          .build();
+
+  public static final ConfigOption<Boolean>
+      KUBERNETES_CLIENT_METRICS_GROUP_BY_RESPONSE_CODE_GROUP_ENABLED =
+          ConfigOption.<Boolean>builder()
+              .key("spark.kubernetes.operator.metrics.clientMetricsGroupByResponseCodeEnabled")
+              .description(
+                  "When enabled, additional metrics group by http response code group(1xx, "
+                      + "2xx, 3xx, 4xx, 5xx) received from API server will be added. Users "
+                      + "can disable it when their monitoring system can combine lower level "
+                      + "kubernetes.client.http.response.<3-digit-response-code> metrics.")
+              .defaultValue(true)
+              .build();
+
+  public static final ConfigOption<Integer> OPERATOR_METRICS_PORT =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.metrics.port")
+          .defaultValue(19090)
+          .description("The port used for checking metrics")
+          .typeParameterClass(Integer.class)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> OPERATOR_PROBE_PORT =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.probePort")
+          .defaultValue(18080)
+          .description("The port used for health/readiness check probe status.")
+          .typeParameterClass(Integer.class)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> SENTINEL_EXECUTOR_SERVICE_POOL_SIZE =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.sentinelExecutorPoolSize")
+          .description(
+              "Size of executor service in Sentinel Managers to check the health "
+                  + "of sentinel resources.")
+          .defaultValue(3)
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .build();
+
+  public static final ConfigOption<Long> SENTINEL_RESOURCE_RECONCILIATION_DELAY =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.health.sentinelResourceReconciliationDelaySeconds")
+          .defaultValue(60L)
```

Comment:
For this discussion (https://github.com/apache/spark-kubernetes-operator/pull/12/files#r1641242260), let's use `Integer` in this case to give a better semantic domain for this configuration.